server/roomlogs.ts

```diff
@@ -211,8 +217,25 @@ export class Roomlog {
 				break;
 			}
 		}
-		this.broadcastBuffer += message + '\n';
-		return this;
+	}
+	addUserUhtmlchange(user: User, name: string, message: string) {
+		const start = `/uhtml ${name},`;
+		const fullMessage = this.addTimestamp(`|c|${user.getIdentity()}|${start}${message}`);
+		for (const [i, line] of this.log.entries()) {
+			if (this.parseChatLine(line)?.message.startsWith(start)) {
+				this.log[i] = fullMessage;
+				break;
+			}
+		}
+		this.broadcastBuffer += fullMessage + '\n';
+	}
+	private parseChatLine(line: string) {
```

Comment:
Considering it's only used for the username at the moment, I would make it do only that. No need to make this more complicated than it needs to be.

Reply:
I wanted to avoid duplication of code from `clearText` and `addUserUhtmlchange`, considering it's a rather complicated code. Do you have a better idea than to do this?

Reply:
Like I said, I would personally replace this with a `getUsernameFromChatLine` function (because that's its only use-case), but I suppose it's unlikely to have a noticeable performance impact. Do what you like.